lib/services/project-changes-service.ts

```diff
@@ -218,20 +219,20 @@ export class ProjectChangesService implements IProjectChangesService {
 	}
 
 	private containsNewerFiles(dir: string, skipDir: string, projectData: IProjectData, processFunc?: (filePath: string, projectData: IProjectData) => boolean): boolean {
+		const dirFileStat = this.$fs.getFsStats(dir);
+		if(this.isFileModified(dirFileStat, dir)) {
+			return true;
+		}
+
 		let files = this.$fs.readDirectory(dir);
 		for (let file of files) {
 			let filePath = path.join(dir, file);
 			if (filePath === skipDir) {
 				continue;
 			}
 
-			let fileStats = this.$fs.getFsStats(filePath);
-
-			let changed = fileStats.mtime.getTime() >= this._outputProjectMtime || fileStats.ctime.getTime() >= this._outputProjectCTime;
-			if (!changed) {
-				let lFileStats = this.$fs.getLsStats(filePath);
-				changed = lFileStats.mtime.getTime() >= this._outputProjectMtime || lFileStats.ctime.getTime() >= this._outputProjectCTime;
-			}
+			const fileStats = this.$fs.getFsStats(filePath);
```

Comment:
the `fileStats` object is not used anywhere outside of the `isFileModified` method. So I believe its place is inside it. At the moment on line 222 we have:
1. get fstat 
2. call `isFileModified`

Here we are doing the same:
1. get fstat
2. call `isFileModified`

I suggest passing only the path to `isFileModified` and getting the fstat in the method itself